### DISCUSSION TITLE
templating name for file - how to use author et al. + title?

### DISCUSSION THREAD

**Discussion by danieltomasz** [aidenlx/obsidian-zotero#120] [Help] [answered]
Really nice plugin! Thank you for your work!
I have a question - does it is possible  to have a  name of literature note in the form
author (et al. if more than 1 or 2 authors) - year - title ?

**Comment by aidenlx** [comment, marked as answer]
I'll add a helper `authorsShort` that output authors in the format you've mentioned in next release, meanwhile you can do something like this: 
```ejs
<%= it.authors.first()?.lastName %>
<% if (it.authors.length > 1) { %> et al. <% } %>
- <%= it.date %> - <%= it.title %>
```

**Reply by danieltomasz** [reply]
Thanks for the advice! While trying to use this I am getting error related to how to get first element of the authors list
`VM668:5 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'first')`

**Reply by aidenlx** [reply]
does replacing `authors` with `creators` work for you?

**Reply by danieltomasz** [reply]
yes, it is working, thank you!

**Reply by lwq-star** [reply]
it does not working
<%= it.title %>_<%= it.creators.first() %>_<%= it.date %> .md
why?

**Comment by aidenlx** [comment]
`zt.authorsShort` exists in v2 for exactly this format (`Smith et al.`). One caveat: the filename template receives item fields plus `zt.creators`, not the full note context, so build the short form from creators there:

```liquid
{% assign first = zt.creators | first %}{{ first.family }}{% if zt.creators.size > 1 %} et al.{% endif %} - {{ zt.date.year }} - {{ zt.title }}
```

Inside the note body `{{ zt.authorsShort }}` gives you the short form directly.

v2 beta is available now: https://zotlit.aidenlx.site/docs/install-zotlit
Migration guide: https://zotlit.aidenlx.site/docs/how-to/migrate-from-v1
